### DISCUSSION TITLE
Simple message

### DISCUSSION THREAD

**Discussion by Tiritibambix** [Opisek/luna#1] [General]
Just a message to show you my support.
I can't wait to try out Luna when docker support will be available.
I wish I could help but I'm already on too many fronts to start learning how to code.
Cheers

**Comment by Opisek** [comment]
Thank you so much for the encouraging words!

Perhaps a short status update on my part:

The development is going fairly smoothly at the moment and I make sure to at least put two hours of coding per day to the project.

Currently I am focused on getting the API working with caldav-based calendar sources. I've recently also connected the basic frontend I had developed earlier to said API and was delighted to see all my events displayed :)

Eventually Luna will support adding multiple "source" types; you'll be able to import calendars from caldav servers, simple iCal links, Google calendar, and even other Luna instances! (That is, someone using Luna can generate a token for your instance to access some calendars stored on their servers. Call it a kind of "federation".)

The top priority currently is getting all my endpoints working with caldav first. I'm running into some hurdles with the available caldav packages.

In particular, there is no way for me to create or edit calendars as such (not the events contained inside) - only fetch them. For now I'm ignoring the problem, but it might stagnate development later on, as I contribute to upstream repositories or write my own caldav implementation (at least for the features that the ready packages all lack).

That being said, I'm not even sure if adding and editing calendars via caldav is standardized. Worst case scenario, the user will be given the choice of adding a fully supported caldav server (development works on Radicale) or a generic caldav server with limited support.

I'm not sure about the timeline that Luna can even begin resembling something production-ready, but I am happy with the steady progress. Ironically, it already lets me do something that Google calendar does not, namely view my caldav calendars from my PC's browser :)

Thanks again for the support! I'll let you know when I reach new milestones. Once I've got enough to show for it, I might reuse my blog's functionality for a Luna devlog.

**Reply by Tiritibambix** [reply]
Thanks for the detailed answer.
Best of luck with this project, I'll be following :)

**Reply by CarlSinclair** [reply]
My two cents: I think there's still value in a read-only calendar UI. There's no reason not to add the ability to create and edit calendar entries over time.

Also looking forward to trying it out, but not with Docker. I'll have a look when I have time and see if I can perhaps contribute.

**Reply by Opisek** [reply]
I agree. I'll first make sure to implement functionality that's straightforward to add. Next on the list is manipulating events.

Both Docker and baremetal installation will be supported. I had just switched development to Docker, because it simplifies a few things for me. Still, I used baremetal before with Makefiles that I kept in the repository. In time, I'm going to document how you can build from source and run without Docker.

You are right, however, that an "unfinished" Luna can already be useful to some people. I might polish up the frontend and release a very early alpha for not much more than read-only purposes once I feel confident enough about it.

**Comment by Tiritibambix** [comment]
Hello ! Still following the pregress.
Any ETA on a prebuilt docker image ?

**Reply by Opisek** [reply]
It is a quite large project so it will not be finished anytime soon.

I would love to release an early preview version before the end of this year, but I'm not 100% certain I can keep that deadline.

**Reply by Opisek** [reply]
@tiritibambix By the way, you can follow the progress more closely by viewing the kanban board linked in the `README.md`. I want to resolve everything from "Current Focus" and "TODO" before releasing a preview version. The "Backlog" bucket is more long-term.

The progress is a bit slower currently, since I am busy with university again, but so far I've been able to put aside some time daily for the project anyway. I try to finish at least one item from the kanban board every day.

**Reply by Tiritibambix** [reply]
Haha Nice, vikunja is the way to go :)

Thank you for mentioning this. I'll be following closely.

Good luck with university 💪
